sdk/personalizer/ai-personalizer-rest/samples/v1-beta/javascript/rankActionsAndRewardEvents.js

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @summary Demonstrates the use of a Personalizer client to rank actions and reward the presented action.
+ */
+const Personalizer = require("@azure-rest/ai-personalizer").default;
+
+// Load the .env file if it exists
+require("dotenv").config();
+
+async function main() {
+  const endpoint = process.env.PERSONALIZER_ENDPOINT || "<endpoint>";
+  const key = process.env.PERSONALIZER_API_KEY || "<test-key>";
+
+  const client = Personalizer(endpoint, { key: key });
+
+  // The list of actions (videos in this case) to be ranked with metadata associated for each action.
+  const actions = [
+    {
+      id: "Video1",
+      features: [
+        { videoType: "documentary", videoLength: 35, director: "CarlSagan" },
+        { mostWatchedByAge: "50-55" },
+      ],
+    },
+    {
+      id: "Video2",
+      features: [
+        { videoType: "movie", videoLength: 120, director: "StevenSpielberg" },
+        { mostWatchedByAge: "40-45" },
+      ],
+    },
+  ];
+
+  // The current context
+  const contextFeatures = [
+    { Features: { day: "tuesday", time: "night", weather: "rainy" } },
+    {
+      Features: {
+        payingUser: true,
+        favoriteGenre: "documentary",
+        hoursOnSite: 0.12,
+        lastWatchedType: "movie",
+      },
+    },
+  ];
+  const request = {
+    actions: actions,
+    contextFeatures: contextFeatures,
+  };
+
+  console.log("Sending rank request");
+  const rankResponse = await client.path("/rank").post({ body: request });
+  if (rankResponse.status != "201") {
+    const error = rankResponse.body;
+    throw error.error;
+  }
+  const rankOutput = rankResponse.body;
+  const eventId = rankOutput.eventId;
+  console.log(
```

Comment:
I think this should be using some logger library, which will by default use console log but will help customer integration with SDK if you have other logger configuration.